src/appWithTranslation.tsx

```diff
@@ -23,20 +23,19 @@ export const appWithTranslation = <Props extends AppProps = AppProps>(
 ) => {
   const AppWithTranslation = (props: Props) => {
     const { _nextI18Next } = props.pageProps as SSRConfig
-    const locale: string | null = _nextI18Next?.initialLocale ?? null
+    let locale: string | null = _nextI18Next?.initialLocale
+                              ?? props?.router?.locale
 
     // Memoize the instance and only re-initialize when either:
     // 1. The route changes (non-shallowly)
     // 2. Router locale changes
+    // 3. UserConfig override changes
     const i18n: I18NextClient | null = useMemo(() => {
-      if (!_nextI18Next) return null
+      if (!_nextI18Next && !configOverride) return null
 
-      let { userConfig } = _nextI18Next
-      const { initialI18nStore } = _nextI18Next
-      const resources =
-        configOverride?.resources ? configOverride.resources : initialI18nStore
+      let userConfig = _nextI18Next?.userConfig
```

Comment:
Maybe something like this would be cleaner?

```suggestion
      let userConfig = configOverride ?? _nextI18Next?.userConfig
```